### PR TITLE
Feature/yash/renter candidate

### LINKED ISSUE #35: Simple Process Tracking: Renter Candidates
# Overview

### Renter Candidates API

Create a `getRenterCandidates` function in `frontend/src/api/renter-candidates.ts` that calls `GET /api/renter-candidates` and create a `createRenterCandidate` function that calls `POST /api/renter-candidates` appropriately.

### Renter Candidate Model

Implement the Renter Candidate model ([schema](https://docs.google.com/document/d/1ef1bwMTZANkUs5uyz-D93v85jv7CfmoDGwE5ANlX_BE/edit#bookmark=id.77oihdu4yyj0)).

### Renter Candidate Routes

Implement the `POST /api/renter-candidates` and `GET /api/renter-candidates` route ([reference](https://docs.google.com/document/d/19EBAxzbCyPdT6RgELrGsq6A2RPeDt2OSi5DYbFWm03k/edit#bookmark=id.dho87e3k6jj4))

## Acceptance Criteria
- [x] Renter Candidate Model
- [x] `POST /api/renter-candidates` route
- [x] `GET /api/renter-candidates` route
- [x] `createRenterCandidate` API route function
- [x] `getRenterCandidates` API route function

### PR DESCRIPTION
**Pull request by YashRavipati1**
## Tracking Info

Resolves #35 

## Changes

Created Renter Candidate functionality in API and made frontend functions to get RenterCandidates and create RenterCandidates.

## Testing

Used frontend functions on the login page and printed responses. Checked mongoDB for correct functionality (renter is created, correct renter info is returned).

## Confirmation of Change

This is the result of using the frontend function createRenterCandidate on the login page.
Screenshot 2024-02-14 at 9 27 52 PM

### CODE REVIEW

File: backend/src/models/renter.ts
Comment (petabite) [inline backend/src/models/renter.ts]: Sorry, this is totally my fault! I forgot to update the schema to reflect the design updates (referring to [this design](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=1186%3A14552&mode=dev)). The schema should up to date now. Do you mind updating to match? 🙏🏼

File: backend/src/controllers/renter.ts
Comment (petabite) [inline backend/src/controllers/renter.ts]: Isn't this equivalent to:

```suggestion
const response = await createRenterCandidate(firstName, lastName, contactInfo, program);
```

File: backend/src/controllers/renter.ts
Comment (petabite) [inline backend/src/controllers/renter.ts]: I think we can just return the whole object from mongo (which includes the `createdAt`, `updatedAt` fields):

```suggestion
res.status(200).json(response);
```

**Review by petabite — CHANGES_REQUESTED** [review]
Great work!! Left some comments. Apologies, one of them is my bad 😬

**Comment by petabite** [comment]
Also, don't worry about the failing preview build

File: backend/src/controllers/renter.ts
Comment (PoliteUnicorn) [inline backend/src/controllers/renter.ts:13]: Would this have to be updated too with the new fields?

File: backend/src/models/renter.ts
Comment (PoliteUnicorn) [inline backend/src/models/renter.ts]: Would you like to keep or remove the comments?

File: backend/src/services/renter.ts
Comment (PoliteUnicorn) [inline backend/src/services/renter.ts]: I think these may need to be updated too.

File: backend/src/models/renter.ts
Comment (YashRavipati1) [reply backend/src/models/renter.ts]: Updated!

File: backend/src/controllers/renter.ts
Comment (YashRavipati1) [reply backend/src/controllers/renter.ts]: Updated!

File: backend/src/controllers/renter.ts
Comment (YashRavipati1) [reply backend/src/controllers/renter.ts]: Updated!

File: backend/src/controllers/renter.ts
Comment (YashRavipati1) [reply backend/src/controllers/renter.ts:13]: Updated!

File: backend/src/models/renter.ts
Comment (YashRavipati1) [reply backend/src/models/renter.ts]: Removed the comments

File: backend/src/services/renter.ts
Comment (YashRavipati1) [reply backend/src/services/renter.ts]: Updated!

File: backend/src/validators/renter.ts
Comment (petabite) [inline backend/src/validators/renter.ts]: Oops l think we forgot validation for the new fields

File: frontend/src/api/renter-candidates.ts
Comment (petabite) [inline frontend/src/api/renter-candidates.ts]: Can we update these types too?

**Review by petabite — CHANGES_REQUESTED** [review]
Thanks for updating! It's looking great, just missed a few things. Otherwise, it should be good to go after the merge conflict is resolved 🎉

File: backend/src/validators/renter.ts
Comment (YashRavipati1) [reply backend/src/validators/renter.ts]: fixed

File: frontend/src/api/renter-candidates.ts
Comment (YashRavipati1) [reply frontend/src/api/renter-candidates.ts]: fixed

**Review by petabite — APPROVED** [review]
Amazing! 🔥 Thanks for the changes!

**Review by PoliteUnicorn — APPROVED** [review]
I think that everything looks good!

**Review by PoliteUnicorn — APPROVED** [review]
I think that everything looks good!

**Pull request merged**
